packages/hardhat-viem/src/types.ts

```diff
@@ -34,18 +43,12 @@ export interface DeployContractConfig extends SendTransactionConfig {
 export type SendDeploymentTransactionConfig = SendTransactionConfig;
 
 export interface GetContractAtConfig {
-  walletClient?: WalletClient;
-  publicClient?: PublicClient;
+  client?: KeyedClient;
 }
 
 export type GetContractReturnType<
   TAbi extends viemT.Abi | readonly unknown[] = viemT.Abi
-> = viemT.GetContractReturnType<
```

Comment:
My understanding of this line is, previously we needed to re-export the `GetContractReturnType` to allow some abi shenanigans (which makes sense as we encode abi's as `any[]`), and to pass explicit `PublicClient` and `WalletClient` over the `viem` type. We did this because the type we get back should have both the public and wallet actions as we create default instances of both.
With `viem2` we change that to match the new viem KeyedType but with both populated. `Required<KeyedClient>` being the equivalent of `{public: PublicClient, wallet: WalletClient}`.